### PR TITLE
keys, runtime: add ed25519 generate, sign, verify

### LINKED ISSUE #357: Keys: generate, encode, decode
We are going to need a package to generate and manage keys. Looks like substrate uses both ed25519 and sr25519 keypairs: https://github.com/paritytech/substrate/blob/a2df800227c6cf11cdb1a642de22ccfea56bf899/core/keystore/src/lib.rs
https://github.com/paritytech/substrate/tree/a2df800227c6cf11cdb1a642de22ccfea56bf899/core/primitives/src

We should create functionality to generate new key pairs, encode existing keys, and decode previously generated keys.

We will also need signing/verification of messages using these keypairs.

This is needed for completion of ext_ functions (generate, sign, verify)

### PR DESCRIPTION
**Pull request by noot**
## Changes
- create keys package, add generate, sign, verify functions for ed25519
- update ext_ed25519_verify to use this package

note that the other ext_ functions for ed25519 require a keystore so I will make an issue for that

## Tests:
```
go test ./crypto/... ./runtime/...
```

### Issues:
closes #357

### CODE REVIEW

File: keys/ed25519.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package keys
```
Comment (ansermino) [inline keys/ed25519.go]: would it make sense for this to be under `crypto`? I'm unsure what the difference is

File: keys/ed25519.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	ed25519 "golang.org/x/crypto/ed25519"
+)
+
+type Ed25519Keypair struct {
+	public  ed25519.PublicKey
+	private ed25519.PrivateKey
+}
+
+func NewEd25519Keypair(priv ed25519.PrivateKey) *Ed25519Keypair {
```
Comment (ansermino) [inline keys/ed25519.go]: Please briefly document these

File: keys/ed25519.go (hunk 1 above)
Comment (noot) [reply keys/ed25519.go]: yeah it could be, that probably makes more sense. I will move it

File: keys/ed25519.go (hunk 2 above)
Comment (noot) [reply keys/ed25519.go]: done

**Pull request merged**